osu.Game/Online/Chat/MessageNotifier.cs

```diff
@@ -129,7 +129,16 @@ private void checkForMentions(Channel channel, Message message)
         /// Checks if <paramref name="message"/> contains <paramref name="username"/>.
         /// This will match against the case where underscores are used instead of spaces (which is how osu-stable handles usernames with spaces).
         /// </summary>
-        private static bool checkContainsUsername(string message, string username) => message.Contains(username, StringComparison.OrdinalIgnoreCase) || message.Contains(username.Replace(' ', '_'), StringComparison.OrdinalIgnoreCase);
+        // Converted the message string to an array of strings split by (' ') and checked if the message contains the username of the user , an attempt to fix  issue #15693
+        private static bool checkContainsUsername(string message, string username)
+        {
+            string[] messageArr = message.Split(' ');
```

Comment:
This will only split by this specific ASCII space character. There are more characters than the `0x20` space that can represent a word break. It would be actually better to pass `null` to `Split()`, as [per docs](https://docs.microsoft.com/en-us/dotnet/api/system.string.split?view=net-6.0#System_String_Split_System_Char___System_StringSplitOptions_):

> If the separator parameter is `null` or contains no characters, white-space characters are assumed to be the delimiters. White-space characters are defined by the Unicode standard and the `Char.IsWhiteSpace` method returns true if they are passed to it.

Also this should probably specify `StringSplitOptions.RemoveEmptyEntries`.

Also would propose to rename to `messageWords` or similar.